File: js/snippets/google-cloud/storage.js
// npm install --save @google-cloud/storage
const {Storage} = require('@google-cloud/storage');

let format = require('string-format'),
    path = require('path');

let list = (bucket) => {
    bucket.getFiles((err, files) => {
        if (err) {
            console.log('Error listing bucket:', err);
        } else {
            files.forEach((f) => {
                //console.log(JSON.stringify(f, null, 4));
                id = f.metadata;
                console.log(format('\nBucket: {}\nName: {}\nContent-Type: {}\nSize: {}\nStorageClass: {}\nCreated: {}',
                    id.bucket, id.name, id.contentType, id.size, id.storageClass, id.timeCreated))
                console.log(format('MD5: {}\nCRC32C: {}\nMediaLink: {}\nACL: {}',
                    id.md5Hash, id.crc32c, id.mediaLink, JSON.stringify(f.parent.storage.acl, null, 4)))
            });
        }
    });
}

let read = (bucket, filename) => {
    let data = '';
    let remoteReadStream = bucket.file(filename).createReadStream();
    remoteReadStream
        .on('data', function (chunk) {
            data += chunk.toString();
            //console.log('CHUNK:', chunk);
        })
        .on('end', function () {
            console.log(data);
        });
}

let write = (bucket, filename) => {
    console.log('TODO');
}

let remove = (bucket, filename) => {
    console.log('TODO');
}

if (process.argv.length < 4) {
    console.log('\nUsage: %s key.json <action> <params>', path.basename(process.argv[0]));
    console.log(`
    create <bucket>
    list <bucket>
    attr <bucket> [<object>]
    read <bucket> <object>
    write <bucket> <object>
    delete <bucket> <object>
    `)
    process.exit(1);
}
// explicitly specifying service_account credentials file
let keyFile = process.argv[2];
const storage = new Storage({
    keyFilename: keyFile
});

let keyData = require(keyFile);
let projectID = keyData.project_id;

let bucket = storage.bucket(process.argv[4]);
switch (process.argv[3]) {
    case "list":
        list(bucket);
        break;
    case "read":
        read(bucket, process.argv[5]);
        break;
    case "write":
        write(bucket, process.argv[5]);
        break;
    case "delete":
        remove(bucket, process.argv[5]);
        break;
}

File: python/snippets/google-cloud/storage.py

# pip install --upgrade google-cloud-storage
from google.cloud import storage

import os
import sys

def create_bucket(bucket_name, blob_name):
    storage_client = storage.Client()
    bucket = storage_client.create_bucket(bucket_name)
    print('Bucket {} created'.format(bucket.name))

def list_blobs(bucket_name, blob_name):
    storage_client = storage.Client()
    bucket = storage_client.get_bucket(bucket_name)
    blobs = bucket.list_blobs()
    for blob in blobs:
        print(blob.name)

def read_blob(bucket_name, blob_name):
    storage_client = storage.Client()
    bucket = storage_client.get_bucket(bucket_name)
    blob = bucket.blob(blob_name)
    blob.download_to_filename(sys.stdout)

def write_blob(bucket_name, blob_name):
    storage_client = storage.Client()
    bucket = storage_client.get_bucket(bucket_name)
    blob = bucket.blob(blob_name)
    blob.upload_from_filename(sys.stdin)

def delete_blob(bucket_name, blob_name):
    storage_client = storage.Client()
    bucket = storage_client.get_bucket(bucket_name)
    blob = bucket.blob(blob_name)
    blob.delete()
    print('Blob {} deleted.'.format(blob_name))

if __name__ == '__main__':
    if len(sys.argv) < 4:
        print '''\nUsage: %s key.json <action> <params>

        create <bucket>
        list <bucket>
        read <bucket> <blob>
        write <bucket> <blob>
        delete <bucket> <blob>
        ''' % os.path.basename(sys.argv[0])
        sys.exit(1);

    os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = sys.argv[1]
    action = sys.argv[2]
    bucket_name = sys.argv[3]
    switcher = {
        'create': create_bucket,
        'list': list_blobs,
        'read': read_blob,
        'write': write_blob,
        'delete': delete_blob
    }
    func = switcher.get(action, 'Invalid Action')
    blob_name = '' if len(sys.argv) < 4 else sys.argv[4]
    func(bucket_name, blob_name);
    # using python 2.7 the google-cloud credentials do NOT appear to be working
